src/algebraic_geometry/open_immersion.lean

```diff
@@ -276,4 +277,84 @@ end
 
 end PresheafedSpace.is_open_immersion
 
+namespace SheafedSpace.is_open_immersion
+
+variable [has_products C]
+
+section prod
+
+variables {ι : Type v} (F : discrete ι ⥤ SheafedSpace C) [has_colimit F] [has_limits C] (i : ι)
+
+lemma sigma_ι_open_embedding : open_embedding (colimit.ι F i).base :=
```

Comment:
I think it would be worthwhile to break this up into the following steps:
1. If `f` is a morphism of topological spaces and `e` is an isomorphism then `open_immersion (e \gg f)` is equivalent to `open_immersion f` (and similarly for the composition in the other direction. These could be simp lemmas.
2. Prove the analogue of the lemma in question for topological spaces.
3. Use the fact that `SheafedSpace.forget` preserves colimits to show that `(colimit.\iota F i).base` is `colimit.\iota` on the level of topological spaces, up-to composition on either side with an isomorphism.
4. Use 1 and 2 to obtain the result. 

This would add some useful lemmas for topological spaces, and would shorten this proof somewhat.

Can the other two long-ish proofs below be broken up similarly?

Reply:
I don't think 1 is needed as a simp lemma since the Lean could already achieve this via instance inferring.
I think the hassle stems from the problem that `open_embedding` is not a typeclass.
Is there a particular reason why these defs (`continuous`, `inducing`, `open_embedding`, etc) are not typeclasses? I think it would be particularly useful in these cases.

Reply:
I think the other long-ish proofs are as short as it could get. The proofs themselves are straightforward, just that every type signature takes up two to three full lines.